src/stirling/source_connectors/socket_tracer/protocols/mqtt/parse.cc

```diff
@@ -0,0 +1,774 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <iomanip>
+#include <memory>
+#include <string_view>
+
+#include "src/common/base/base.h"
+#include "src/stirling/source_connectors/socket_tracer/protocols/mqtt/types.h"
+#include "src/stirling/utils/binary_decoder.h"
+#include "src/stirling/utils/parse_state.h"
+
+#define PX_ASSIGN_OR_RETURN_NEEDS_MORE_DATA(expr, val_or) \
+  PX_ASSIGN_OR(expr, val_or, return ParseState::kNeedsMoreData)
+#define PX_ASSIGN_OR_RETURN_INVALID(expr, val_or) \
+  PX_ASSIGN_OR(expr, val_or, return ParseState::kInvalid)
+namespace px {
+namespace stirling {
+namespace protocols {
+
+namespace mqtt {
+
+enum class MqttControlPacketType : uint8_t {
+  CONNECT = 1,
+  CONNACK = 2,
+  PUBLISH = 3,
+  PUBACK = 4,
+  PUBREC = 5,
+  PUBREL = 6,
+  PUBCOMP = 7,
+  SUBSCRIBE = 8,
+  SUBACK = 9,
+  UNSUBSCRIBE = 10,
+  UNSUBACK = 11,
+  PINGREQ = 12,
+  PINGRESP = 13,
+  DISCONNECT = 14,
```

Comment:
Is there a reason you didn't want to add the `AUTH` control packet type? It seems like that is the only one omitted from this list.

Reply:
I forgot to include this, sorry about this. I have added this with tests in the new commit.